### PR TITLE
Converts the space heater to tgui

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Converts the space heater to tgui

## Why It's Good For The Game
browser UI bad for this

## Images of changes
Screenshot 2023-11-29 190307

## Testing
opened the tgui menu

## Changelog
:cl:
tweak: Space heater ui is now TGUI
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/SpaceHeater.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          <LabeledList.Item label="Cell Charge">
+            {data.CellPercent}%
+          </LabeledList.Item>
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const GetCellButtons = (props, context) => {
```
Comment (Gaxeer) [inline tgui/packages/tgui/interfaces/SpaceHeater.js:49]: As you can see, other components have names as nouns. And this component always returns just one `button`.
So, the better name in my opinion will be just `CellButton`

File: code/game/machinery/spaceheater.dm [hunk 2]
```diff
@@ -82,85 +82,76 @@
...
-		usr << browse(null, "window=spaceheater")
-		usr.unset_machine()
-	return
+	switch(action)
+		if("change_temp")
+			set_temperature = dd_range(0, 90, text2num(params["change_temp"]))
+		if("add_cell")
+			if(!open || cell)
+				return
+			var/obj/item/stock_parts/cell/C = usr.get_active_hand()
```
Comment (Gaxeer) [inline code/game/machinery/spaceheater.dm:131]: One letter var names are bad

File: code/game/machinery/spaceheater.dm [hunk 3]
```diff
@@ -82,85 +82,76 @@
...
+	. = TRUE
 
-		updateDialog()
-	else
-		usr << browse(null, "window=spaceheater")
-		usr.unset_machine()
-	return
+	switch(action)
+		if("change_temp")
+			set_temperature = dd_range(0, 90, text2num(params["change_temp"]))
```
Comment (Gaxeer) [inline code/game/machinery/spaceheater.dm:127]: 0 and 90 - magic numbers

File: code/game/machinery/spaceheater.dm [hunk 4]
```diff
@@ -82,85 +82,76 @@
...
+			usr.visible_message("<span class='notice'>[usr] inserts a power cell into [src].</span>", "<span class='notice'>You insert the power cell into [src].</span>")
+		if("remove_cell")
+			if(!open || !cell || usr.get_active_hand())
+				return
+			on = FALSE
+			cell.update_icon()
+			usr.put_in_hands(cell)
+			cell.add_fingerprint(usr)
+			cell = null
+			usr.visible_message("<span class='notice'>[usr] removes the power cell from [src].</span>", "<span class='notice'>You remove the power cell from [src].</span>")
```
Comment (Gaxeer) [inline code/game/machinery/spaceheater.dm:148]: Will be good to move `remove_cell` and `add_cell` to separate procs, for readability

File: tgui/packages/tgui/interfaces/SpaceHeater.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+              onClick={() => act('toggle_power')}
+            />
+          </LabeledList.Item>
+          <LabeledList.Item label="Cell">
+            <GetCellButtons />
+          </LabeledList.Item>
+          <LabeledList.Item label="Cell Charge">
+            {data.CellPercent}%
+          </LabeledList.Item>
+        </Section>
```
Comment (Gaxeer) [inline tgui/packages/tgui/interfaces/SpaceHeater.js:43]: Move to separate component, please

File: code/game/machinery/spaceheater.dm [hunk 6]
```diff
@@ -82,85 +82,76 @@
...
-	else
-		usr << browse(null, "window=spaceheater")
-		usr.unset_machine()
-	return
+	switch(action)
+		if("change_temp")
+			set_temperature = dd_range(0, 90, text2num(params["change_temp"]))
+		if("add_cell")
+			if(!open || cell)
+				return
```
Comment (Gaxeer) [inline code/game/machinery/spaceheater.dm:130]: You set . to `TRUE`, which means, that UI update will be performed. But in this case, it's needed. As I can see.
So `return FALSE`

File: code/game/machinery/spaceheater.dm [hunk 7]
```diff
@@ -82,85 +82,76 @@
...
+			var/obj/item/stock_parts/cell/C = usr.get_active_hand()
+			if(!istype(C))
+				return
```
Comment (Gaxeer) [inline code/game/machinery/spaceheater.dm:133]: also `return FALSE`

File: code/game/machinery/spaceheater.dm [hunk 8]
```diff
@@ -82,85 +82,76 @@
...
+			usr.drop_item()
+			cell = C
+			C.forceMove(src)
+			C.add_fingerprint(usr)
+
+			usr.visible_message("<span class='notice'>[usr] inserts a power cell into [src].</span>", "<span class='notice'>You insert the power cell into [src].</span>")
+		if("remove_cell")
+			if(!open || !cell || usr.get_active_hand())
+				return
```
Comment (Gaxeer) [inline code/game/machinery/spaceheater.dm:142]: `return FALSE`

**Review by S34NW — CHANGES_REQUESTED** [review]
Please see reviews

**Comment by S34NW** [comment]
Stale PR, no changes made in requested timeframe. Please re-open this once changes are made or make a new PR.

**Pull request closed without merging**